src/app/qgsgeometryvalidationservice.cpp

```diff
@@ -346,11 +346,17 @@ void QgsGeometryValidationService::invalidateTopologyChecks( QgsVectorLayer *lay
 
 void QgsGeometryValidationService::processFeature( QgsVectorLayer *layer, QgsFeatureId fid )
 {
+  if ( !mLayerChecks.contains( layer ) )
+    return;
+
+  const QList< QgsSingleGeometryCheck * > checks = mLayerChecks[layer].singleFeatureChecks;
+  if ( checks.empty() )
+    return;
+
   emit geometryCheckStarted( layer, fid );
 
   QgsGeometry geometry = layer->getGeometry( fid );
 
-  const auto &checks = mLayerChecks[layer].singleFeatureChecks;
```

Comment:
This is a great example of how the use of auto makes it much harder to glance over the code.